lib/ssh/src/ssh_acceptor_subsup.erl

```diff
@@ -0,0 +1,142 @@
+    Self = self(),
+    NumAcceptors = case ?GET_OPT(parallel_login, Options) of
+                       true -> 1;
+                       false -> 1;
+                       N -> N
+                   end,
+    {Sock, Opts} = case ?GET_INTERNAL_OPT(lsocket, Options, undefined) of
+        {LSock, SockOwner} ->
+            %% A listening socket (or fd option) was provided in the ssh:daemon call
+            case inet:sockname(LSock) of
+                {ok, {_, Port}} ->
+                    %% A usable, open LSock
+                    spawn_link(fun() ->
+                              SockOwner ! {request_control, LSock, Self, self()},
+                              receive {its_yours, LSock} -> ok end,
+                              [supervisor:start_child(Self, []) || _ <- lists:seq(1, NumAcceptors)],
+                              unlink(Self)
+                          end),
+                    {LSock, Options};
+
+                {error, _Error} ->
+                    %% Not open, a restart
+                    %% Allow gen_tcp:listen to fail 4 times if eaddrinuse (It is a bug fix):
+                    case try_listen(Port, Options, 4) of
+                        {ok, NewLSock} ->
+                            spawn_link(fun() ->
+                                           [supervisor:start_child(Self, []) || _ <- lists:seq(1, NumAcceptors)],
+                                           unlink(Self)
+                                       end),
+                            {NewLSock, ?DELETE_INTERNAL_OPT(lsocket, Options)};
+                        {error, Error} ->
+                            exit({error, Error})
+                    end
+            end;
+
+        undefined ->
+            %% No listening socket (nor fd option) was provided; open a listening socket:
+            case listen(Port, Options) of
+                {ok, LSock} ->
+                    spawn_link(fun() ->
+                                   [supervisor:start_child(Self, []) || _ <- lists:seq(1, NumAcceptors)],
+                                   unlink(Self)
+                               end),
+                    {LSock, Options};
+                {error, Error} ->
+                    exit({error, Error})
+            end
+    end,
+    SupFlags = #{strategy  => simple_one_for_one, 
+                 intensity => 1000000,
+                 period    => 1
+                },
+    ChildSpecs = [#{id => ssh_acceptor,
+                    start => {ssh_acceptor, start_link, [SystemSup, Sock, Address, Opts]},
+                    restart => transient}],
+    {ok, {SupFlags,ChildSpecs}}.
+
+%%%=========================================================================
+%%%  Internal functions
+%%%=========================================================================
+
+listen(Port, Options) ->
+    {_, Callback, _} = ?GET_OPT(transport, Options),
+    SockOpts = ?GET_OPT(socket_options, Options) ++ [{active, false}, {reuseaddr, true}],
```

Comment:
This was `[{active, false}, {reuseaddr,true} | ?GET_OPT(socket_options, Options)]` in `ssh_acceptor`, which is wrong. For one, it would prevent the usage of option `{inet_backend, socket}` since this option must be the first in the list. For another, in options given to `gen_tcp` or `ssl`, the _last_, option wins, such that if for example `{active, true}` was in the list of given socket options, it would be set to `{active, true}` despite the tacked-on `{active, false}`.